integration-tests/utils.py

```diff
@@ -20,50 +21,14 @@
 
 
 @log_calls
-def dump_model_info(model, log_dir):
-    ''' Dumps information about the model to the log dir '''
-    data = {
-        'applications': {k: v.data for k, v in model.applications.items()},
-        'units': {k: v.data for k, v in model.units.items()},
-        'machines': {k: v.data for k, v in model.machines.items()}
-    }
-    path = os.path.join(log_dir, 'model-info')
-    with open(path, 'w') as f:
-        json.dump(data, f, indent=2)
-        f.write('\n')
-
-
-@log_calls_async
-async def dump_debug_log(model, log_dir):
-    ''' Dumps Juju debug log to the log dir '''
-    path = os.path.join(log_dir, 'debug-log')
-    with open(path, 'w') as f:
-        cmd = ['juju', 'debug-log', '-m', model.info.name, '--replay']
-        await asyncify(subprocess.call)(cmd, stdout=f,
-                                        stderr=subprocess.STDOUT)
-
-@log_calls_async
-async def dump_debug_actions(model, log_dir):
-    ''' Runs debug action on all units, dumping the results to log dir '''
-    result_dir = os.path.join(log_dir, 'debug-actions')
-    os.mkdir(result_dir)
-
-    async def dump_debug_action(unit):
-        try:
-            action = await unit.run_action('debug')
-        except JujuError as e:
-            if 'no actions defined on charm' in str(e) \
-                    or 'not defined on unit' in str(e):
-                return
-            raise
-        await action.wait()
-        remote_path = action.data['results']['path']
-        filename = unit.name.replace('/', '_') + '.tar.gz'
-        local_path = os.path.join(result_dir, filename)
-        await scp_from(unit, remote_path, local_path)
-
-    coroutines = [dump_debug_action(unit) for unit in model.units.values() if unit]
-    await asyncio.wait(coroutines)
+def fetch_field_agent_and_run(log_dir):
+    url = 'https://raw.githubusercontent.com/juju-solutions/cdk-field-agent/master/collect.py'
```

Comment:
This means the integration tests will need to be able to reach `raw.githubusercontent.com`. I think we should be able to access it via proxy in our internal testing machines, but I'm not sure. @ktsakalozos do you anticipate any problems with this requirement?